lib/iris/fileformats/nimrod_load_rules.py

```diff
@@ -25,58 +28,306 @@
+            )
+        else:
+            cube.data = cube.data / float(unit_list[1])
+        field_units = unit_list[0]
+    if "ug/m3E1" in field_units:
+        # Split into unit string and integer
+        unit_list = field_units.split("E")
+        cube.data = cube.data / 10.0 ** float(unit_list[1])
+        field_units = unit_list[0]
+    if "%" in field_units:
+        # Convert any percentages into fraction
+        unit_list = field_units.split("%")
+        if len("".join(unit_list)) == 0:
+            field_units = "1"
+            cube.data = cube.data / 100.0
+    if field_units == "oktas":
+        field_units = "1"
+        cube.data /= 8.0
+    if field_units == "dBZ":
+        # cf_units doesn't recognise decibels (dBZ), but does know BZ
+        field_units = "BZ"
+        cube.data /= 10.0
+    if not field_units:
+        if field.field_code == 8:
+            # Relative Humidity data are unitless, but not "unknown"
+            field_units = "1"
+        if field.field_code in [505, 515]:
+            # CAPE units are not always set correctly. Assume J/kg
+            field_units = "J/kg"
+    if field_units in unit_exception_dictionary.keys():
+        field_units = unit_exception_dictionary[field_units]
+    if len(field_units) > 0 and field_units[0] == "/":
+        # Deal with the case where the units are of the form '/unit' eg
+        # '/second' in the Nimrod file. This converts to the form unit^-1
+        field_units = field_units[1:] + "^-1"
     try:
-        cube.units = units
+        cube.units = field_units
     except ValueError:
         # Just add it as an attribute.
         warnings.warn(
-            "Unhandled units '{0}' recorded in cube attributes.".format(units)
+            "Unhandled units '{0}' recorded in cube attributes.".format(
+                field_units
+            )
         )
-        cube.attributes["invalid_units"] = units
+        cube.attributes["invalid_units"] = field_units
 
 
 def time(cube, field):
     """Add a time coord to the cube."""
-    valid_date = cftime.datetime(
-        field.vt_year,
-        field.vt_month,
-        field.vt_day,
-        field.vt_hour,
-        field.vt_minute,
-        field.vt_second,
-    )
-    point = TIME_UNIT.date2num(valid_date)
-
-    bounds = None
-    if field.period_minutes != field.int_mdi and field.period_minutes != 0:
-        # Create a bound array to handle the Period of Interest if set.
-        bounds = (point - (field.period_minutes / 60.0), point)
+    if field.vt_year <= 0:
+        # Some ancillary files, eg land sea mask do not
+        # have a validity time. So make one up for the
+        # start of the year.
+        # This will screw up the forecast_period for these fields,
+        # although if the valid time is missing too, it will be
+        # made to be the same, so the forecast_period will always
+        # be zero for these files.
+        valid_date = cftime.datetime(2016, 1, 1, 0, 0, 0)
```

Comment:
2016 seems a slightly unexpected magic number, when above it says "for the start of the year".
Can we give a reason for this ?

Reply:
I think this is an oversight. There's no need to set a `"time"` coord for ancillary data, so I've replaced this with `return`.